olive/model/__init__.py

```diff
@@ -56,68 +57,61 @@ def __init__(
         framework: Framework,
         model_file_format: ModelFileFormat,
         model_path: OLIVE_RESOURCE_ANNOTATIONS = None,
+        # resources other than model_path
+        resources: Optional[Dict[str, OLIVE_RESOURCE_ANNOTATIONS]] = None,
     ):
         self.framework = framework
         self.model_file_format = model_file_format
-        self.model_resource_path = create_resource_path(model_path) if model_path else None
-        self.local_model_path = None
         self.composite_parent = None
         self.io_config = None
+        # store resource paths
+        self.resource_paths = {}
+        # this is for storing local instances of resource paths
+        self.local_resource_paths = {}
+        resources = resources or {}
+        resources["model_path"] = model_path
+        for resource_name, resource_path in resources.items():
+            self.resource_paths[resource_name] = create_resource_path(resource_path)
```

Comment:
>resource_path [](http://example.com/codeflow?start=70&length=13)

need we check whether the resource_path is None or not?

Reply:
since all resources are optional, we don't need to check for None. `resource_paths` keep tracks of all resources even though the values are None. 
create_resource_path can handle None also